lib/backend/buffer.go

```diff
@@ -352,7 +366,7 @@ func (w *BufferWatcher) emit(e Event) (ok bool) {
 	defer w.bmu.Unlock()
 
 	if !w.flushBacklog() {
-		if w.buffer.cfg.clock.Now().After(w.backlogSince.Add(w.buffer.cfg.gracePeriod)) {
+		if now := w.buffer.cfg.clock.Now(); now.After(w.backlogSince.Add(w.buffer.cfg.gracePeriod)) && now.After(w.created.Add(w.buffer.cfg.creationGracePeriod)) {
```

Comment:
Slightly cheaper to keep `created.Add(creationGracePeriod)` as a `creationGraceDeadline` or `backlogGraceDeadline` perhaps?

Reply:
Not 100% certain I follow this one.  Do you mean, rather than performing the addition during the check, perform the addition once when storing the backlog creation time?  If so, yeah that would be more efficient, tho not meaningfully so I think given that we only use this logic for the cache and we're already doing blocking json serialization per-event.  Optimizing out an add per event seems silly.

Reply:
I was thinking more about the fact that it's a check against a fixed time point and a deadline would represent that better - but the deadline could be changed just like `created` would, anyway.